### PR TITLE
Add new csv and update operator to version 1.2.0

### PR DESCRIPTION
**Pull request by evan-hataishi**
Resolves https://github.com/kubeflow/kfctl/issues/422 

Operator image v1.2.0 built and pushed to [dockerhub](https://hub.docker.com/layers/127115357/aipipeline/kubeflow-operator/v1.2.0/images/sha256-d7b7c112ab5b5e64588f45819a3b43c3e01aa43b843150ba50340d0d994abf0e?context=explore).

### CODE REVIEW

File: deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml
Comment (Tomcli) [inline deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml:274]: @evan-hataishi You can add your name to the list of maintainers here.

**Comment by Tomcli** [comment]
Thanks @evan-hataishi, it looks good to me. Do you also have a PR that updates it on the operatorHub?

**Comment by evan-hataishi** [comment]
@Tomcli I'll look into this now and submit a PR to operatorhub as well.

**Comment by evan-hataishi** [comment]
@Tomcli Added to operatorhub in https://github.com/operator-framework/community-operators/pull/2734

**Comment by adrian555** [comment]
@evan-hataishi at least the `alm-examples` in the csv file needs to update and reflect the kfdef 1.2.0. And test the operator on k8s cluster.

File: deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml
Comment (Tomcli) [inline deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml:20]: scc is for openshift only. If you are pushing this to the community operator, their CI is deploying it on regular kubernetes.

For the openshift version of this operator, the Red Hat team is maintaining it and they call it OpenDataHub.

**Comment by Tomcli** [comment]
Thanks @evan-hataishi 
/lgtm

File: deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml
Comment (adrian555) [inline deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml:191]: Please refer to the comment I have in https://github.com/operator-framework/community-operators/pull/2734 wrt description part. Let's keep them in sync from this version.

Comment (evan-hataishi) [reply deploy/olm-catalog/kubeflow/1.2.0/kubeflow.v1.2.0.clusterserviceversion.yaml:191]: I'll add all fixes to that version then come back to this PR once it's finalized.

**Comment by evan-hataishi** [comment]
This csv has been merged in https://github.com/operator-framework/community-operators/pull/2734, so both repositories should be in sync after this is merged.

Resolves https://github.com/kubeflow/kfctl/issues/422

**Comment by Tomcli** [comment]
thanks @evan-hataishi 
/lgtm

**Review by adrian555 — APPROVED** [review]

**Pull request merged**
